lib/js/components/SelectionTile/SelectionTile.vue

```diff
@@ -0,0 +1,258 @@
+	&.-disabled:hover {
+		background-color: $color-neutral-background-weak-disabled;
+		cursor: initial;
+		outline-color: $color-neutral-border-disabled;
+
+		#{$root}__title {
+			color: $color-neutral-text-heavy-disabled;
+		}
+
+		#{$root}__supportingText {
+			color: $color-neutral-text-disabled;
+		}
+
+		#{$root}__icon {
+			color: $color-neutral-icon-disabled;
+		}
+
+		&.-selected {
+			background-color: $color-primary-background-disabled;
+			outline-color: $color-primary-border-disabled;
+
+			#{$root}__icon {
+				color: $color-primary-icon-disabled;
+			}
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import { defineComponent, PropType, toRaw } from 'vue';
+import {
+	SELECTION_TILE_STATE,
+	SELECTION_TILE_TYPE,
+	SelectionTileState,
+	SelectionTileType,
+} from './SelectionTile.consts';
+import { ICON_SIZES, IconItem, ICONS } from '../Icons/Icon';
+import RadioButton from '../Form/RadioButton/RadioButton.vue';
+import { RADIO_BUTTON_SIZE, RADIO_BUTTON_STATE } from '../Form/RadioButton/RadioButton.consts';
+import Icon from '../Icons/Icon/Icon.vue';
+import Checkbox from '../Form/Checkbox/Checkbox.vue';
+import { CHECKBOX_SIZE, CHECKBOX_STATE } from '../Form/Checkbox/Checkbox.consts';
+
+const RADIO_BUTTON_STATE_MAP = {
+	[SELECTION_TILE_STATE.DEFAULT]: RADIO_BUTTON_STATE.DEFAULT,
+	[SELECTION_TILE_STATE.LOADING]: RADIO_BUTTON_STATE.LOADING,
+	[SELECTION_TILE_STATE.DISABLED]: RADIO_BUTTON_STATE.DISABLED,
+};
+
+const CHECKBOX_STATE_MAP = {
+	[SELECTION_TILE_STATE.DEFAULT]: CHECKBOX_STATE.DEFAULT,
+	[SELECTION_TILE_STATE.LOADING]: CHECKBOX_STATE.LOADING,
+	[SELECTION_TILE_STATE.DISABLED]: CHECKBOX_STATE.DISABLED,
+};
+
+export default defineComponent({
+	name: 'SelectionTile',
+	components: { Checkbox, Icon, RadioButton },
+	props: {
+		type: {
+			type: String as PropType<SelectionTileType>,
+			default: SELECTION_TILE_TYPE.RADIO_BUTTON,
+			validator(type: SelectionTileType) {
+				return Object.values(SELECTION_TILE_TYPE).includes(type);
+			},
+		},
+		title: {
+			type: String,
+			required: true,
+		},
+		supportingText: {
+			type: String,
+			default: null,
+		},
+		icon: {
+			type: Object,
+			default: null,
+			validator(icon: IconItem) {
+				return Object.values(ICONS).includes(toRaw(icon));
+			},
+		},
+		isSelected: {
+			type: Boolean,
+			default: false,
+		},
+		state: {
+			type: String as PropType<SelectionTileState>,
+			default: SELECTION_TILE_STATE.DEFAULT,
+			validator(state: SelectionTileState) {
+				return Object.values(SELECTION_TILE_STATE).includes(state);
+			},
+		},
+	},
+	emits: ['update:isSelected', 'icon-click'],
+	data() {
+		return {
+			ICONS: Object.freeze(ICONS),
+			ICON_SIZES: Object.freeze(ICON_SIZES),
+			RADIO_BUTTON_SIZE: Object.freeze(RADIO_BUTTON_SIZE),
+			SELECTION_TILE_STATE: Object.freeze(SELECTION_TILE_STATE),
+			SELECTION_TILE_TYPE: Object.freeze(SELECTION_TILE_TYPE),
+			CHECKBOX_SIZE: Object.freeze(CHECKBOX_SIZE),
+		};
+	},
+	computed: {
+		selectionControlState() {
```

Comment:
instead of this computed, maybe we should have a universal state for both radio and checkbox?